test/built-ins/Array/fromAsync/not-a-constructor.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.fromasync
+description: Array.fromAsync is not a constructor
+info: |
+  Built-in function objects that are not identified as constructors do not
+  implement the [[Construct]] internal method unless otherwise specified in the
+  description of a particular function.
+includes: [isConstructor.js]
+features: [Array.fromAsync, Reflect.construct]
+---*/
+
+assert(!isConstructor(Array.fromAsync), "Array.fromAsync is not a constructor");
+
+assert.throws(TypeError, () => new Array.fromAsync(), "Array.fromAsync throws when constructed");
```

Comment:
Maybe add `assert.sameValue(typeof Assert.fromAsync, 'function')` just so implementations don't get this one for free.

Reply:
Or maybe [do it once and for all over here](https://github.com/tc39/test262/pull/3748).

Reply:
I agree with the suggestion. Now that that improvement to isConstructor is merged, I think no change is needed here any longer.